test/test_commands.js

```diff
@@ -0,0 +1,111 @@
+var assert = require('chai').assert;
+var commands = require('../src/commands');
+
+describe('commands', function(){
+    describe('#tokenize()', function(){
+        function assertIfCalled(message){
+            return function(){ assert(false, message); };
+        }
+        it("tokenize a constant string as an object containing parameters for each constant, of the same name and with the constant as the value", function(){
+           return commands.tokenize('this is a test string of constants', [
+               'this', 'is', 'a', 'test', 'string', 'of', 'constants'
+           ]).then(function(parameters){
+               assert.deepEqual(parameters, {
+                   'this':'this',
+                   'is':'is',
+                   'a':'a',
+                   'test':'test',
+                   'string':'string',
+                   'of':'of',
+                   'constants':'constants'
+               });
+           });
+        });
+        it("tokenize a command with all tokenDescriptors", function(){
+            return commands.tokenize('constant 4 text option1', [
+                'constant', 
+                '{int:intVar}',
+                '{text:textVar}',
+                '{option1|option2}'
+            ]).then(function(parameters){
+                assert.deepEqual(parameters, {
+                    'constant': 'constant',
+                    'intVar': 4,
+                    'textVar': 'text',
+                    'option1': 'option1',
+                });
+            });
+        });
+        it("throw an InvalidConstantError if the token does not exactly match the specified constant", 
+            function(){
+                return commands.tokenize('this', ['that']).then(assertIfCalled('tokenize did not throw'))
```

Comment:
I would prefer a better error message here. "Did not throw when a constant doesn't not match the input string" or some such.